pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Replicator.java

```diff
@@ -29,9 +29,9 @@ public interface Replicator {
 
     ReplicatorStatsImpl getStats();
 
-    CompletableFuture<Void> disconnect();
+    CompletableFuture<Void> terminate();
 
-    CompletableFuture<Void> disconnect(boolean b);
+    CompletableFuture<Void> terminate(boolean b);
```

Comment:
```suggestion
    CompletableFuture<Void> terminate(boolean failIfHasBacklog);
```

Reply:
This API looks weird. We should not to fail the termination because `failIfHasBacklog`? I suppose `failIfHasBacklog` should be a part of replication disabling, not termination.

Reply:
> This API looks weird. We should not fail the termination because failIfHasBacklog? I suppose failIfHasBacklog should be a part of replication disabling, not termination.

It is only used by `Topic GC` 😂, so it looks weird. 

I reviewed the code, `Topic GC` used `disconnect(true /* failIfHasBacklog */)` to check there is no backlog of Replicator, and deleted the topic after the disconnect was successful. So, it should want to close the remote producer not terminate the Replicator. I added the method `disconnect` back, and the implementation is just close the remote producer.